torch/jit/__init__.py

```diff
@@ -1071,6 +1072,17 @@ def script(obj, optimize=True, _frames_up=0, _rcb=None):
         if not _is_new_style_class(obj):
             raise RuntimeError("TorchScript classes must be new-style classes. Please inherit from 'object'")
         qualified_name = _qualified_name(obj)
+        if issubclass(obj, tuple) and hasattr(obj, '_fields'):
+            fields = list(obj._fields)
+            if hasattr(obj, '__annotations__'):
+                annotations = OrderedDict()
+                for k in obj.__annotations__.keys():
+                    annotations[k] = torch.jit.annotations.ann_to_type(obj.__annotations__[k])
+            else:
+                annotations = OrderedDict()
+            _jit_script_named_tuple_compile(qualified_name, fields, annotations, _rcb)
+            _add_script_class(obj, qualified_name)
```

Comment:
I realized we don't actually need any of these name -> type mappings at all (we don't need the one in CompilationUnit either). Since we're structurally typing NamedTuples, all we need is a "Python namedtuple -> JIT TupleType" translation. When the compiler encounters a type `Foo`, it should just look up `Foo` in Python environment; if it's a named tuple, generate the corresponding tuple type, and use that.